src/main/java/com/courseApp/calendarService/SingledayCalendar.java

```diff
@@ -0,0 +1,63 @@
+package com.courseApp.calendarService;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.courseApp.constants.Constants;
+import com.courseApp.constants.Exceptions;
+import com.courseApp.utils.TimeConversion;
+
+
+public abstract class SingledayCalendar extends CalendarFactory implements UsePresentable {
+    public SingledayCalendar(Map<String, Map<String, ArrayList<String>>> formattedSchedule) {
+        super(formattedSchedule);
+    }
+
+    /**
+     * Reformat the list by marking column that is occupied by day in the week.
+     *
+     * @param list the Calendar list
+     * @param calendarType type of the Calendar required
+     */
+    protected void markSingleDay(List<String> list, String calendarType){
+        list.set(Constants.FIRST_DAY_POSITION, list.get(Constants.FIRST_DAY_POSITION).substring(0,
+                Constants.CELL_FILLING+Constants.COURSE_CODE_FLAG)+
+                calendarType.substring(0, Constants.COURSE_WEEKDAY_FLAG) +
+                list.get(Constants.FIRST_DAY_POSITION).
+                        substring(Constants.CELL_FILLING+Constants.COURSE_CODE_FLAG+Constants.COURSE_WEEKDAY_FLAG));
+
+    }
+    /**
+     * Convert the String weekday to an integer format indicating its location in SingledayCalendar.
+     *
+     * @param weekday String of Days of week
+     * @return weekday's location in the SingledayCalendar
+     */
+    protected int getWeekDay(String weekday){
+//        if (weekday.equals(Constants.MONDAY) | weekday.equals(Constants.TUESDAY)| weekday.equals(Constants.WEDNESDAY)|
+//                weekday.equals(Constants.THURSDAY)| weekday.equals(Constants.FRIDAY)) {
+            return Constants.LOCATION_MO;
```

Comment:
If this constant is unique for all the singledayCalendars, i.e., not only for the monday calendar. Then the name of this constant shall specify it.